identifier/identifier_test.go

```diff
@@ -0,0 +1,76 @@
+package identifier
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net"
+	"net/netip"
+	"testing"
+
+	"github.com/letsencrypt/boulder/test"
+)
+
+func TestNormalize(t *testing.T) {
```

Comment:
This test should include IP-type identifiers, to show that your sorting logic is always putting them after DNS identifiers.

Also, I'd personally say that this would be better as a table-driven test, testing each aspect of Normalize individually:
- does it lowercase?
- does it sort?
- does it deduplicate?